src/backend/distributed/citus--5.1-5--5.1-6.sql

```diff
@@ -0,0 +1,67 @@
+CREATE FUNCTION pg_catalog.master_drop_sequences(sequence_names text[],
+												 node_name text,
+												 node_port bigint)
+    RETURNS bool
+    LANGUAGE C STRICT
+    AS 'MODULE_PATHNAME', $$master_drop_sequences$$;
+COMMENT ON FUNCTION pg_catalog.master_drop_sequences(text[], text, bigint)
+    IS 'drop specified sequences from a node';
+
+REVOKE ALL ON FUNCTION pg_catalog.master_drop_sequences(text[], text, bigint) FROM PUBLIC;
+
+CREATE OR REPLACE FUNCTION pg_catalog.citus_drop_trigger()
+    RETURNS event_trigger
+    LANGUAGE plpgsql
+    SET search_path = pg_catalog
+    /* declared as SECURITY DEFINER in upgrade script */
+    AS $cdbdt$
+DECLARE
+	v_obj record;
+	sequence_names text[] := '{}';
+	node_names text[] := '{}';
+	node_ports bigint[] := '{}';
+	node_name text;
+	node_port bigint;
+BEGIN
+	SELECT array_agg(object_identity) INTO sequence_names
+	FROM pg_event_trigger_dropped_objects()
+	WHERE object_type = 'sequence';
+
+	SELECT array_agg(sp.nodename), array_agg(sp.nodeport)
```

Comment:
Can you add a comment explaining what were doing here, and why it's separate from the loop below?